docker/Dockerfile

```diff
@@ -48,23 +50,20 @@ RUN find /definitions -type d -exec chmod 0775 {} \;
 
 #############################################################
 
-# Build the UBI8 image
-FROM redhat/ubi8-minimal:8.10 AS cass-config-builder-ubi
+# Build the UBI9 image using Eclipse Temurin for Java runtime
+FROM redhat/ubi10-minimal:10.0 AS cass-config-builder-ubi
 LABEL maintainer="DataStax, Inc <info@datastax.com>"
 LABEL name="cass-config-builder"
 LABEL vendor="DataStax, Inc"
 LABEL release="1.0.0"
 LABEL summary="Configuration templating engine for Apache Cassandra®."
 LABEL description="Configuration templating engine for Apache Cassandra®. Powers the configuration of containers deployed via the DataStax Kubernetes Operator for Apache Cassandra."
 
-# Update base packages
-RUN microdnf update && \
-    rm -rf /var/cache/yum && \
-    microdnf install java-1.8.0-openjdk-headless && \
-    microdnf clean all
-
-# Copy user account information
-COPY --from=builder-ubi /etc/passwd /etc/shadow /etc/group /etc/gshadow /etc/
+# Install shadow-utils and create user and group for security
+RUN microdnf update -y --nogpgcheck && \
+    microdnf install -y --nogpgcheck shadow-utils java-21-openjdk-headless && \
+    microdnf clean all && \
+    groupadd -r cassandra && useradd -r -g cassandra cassandra
```

Comment:
I think this needs to be cassandra:root instead of cassandra:cassandra to function correctly in certain securityContexts.